handlers/BES_Modules_The_HDF5_Handler.adoc

```diff
@@ -1,126 +1,468 @@
 = The HDF5 Handler
-:Leonard Porrello <lporrel@gmail.com>:
+:Kent Yang <myang6@hdfgroup.org>:
 {docdate}
 :numbered:
 :toc:
 
 == Introduction
 
-This release of the server supports HDF5 files written using any version
-of the HDF5 API. The handler should be built/linked with version 1.8.x
-of the API.
+HDF5 handler not only supports the general mapping from HDF5 to either DAP2 or DAP4, but can also make NASA HDF5/HDF-EOS5/netCDF-4-like files follow the https://cfconventions.org/[CF conventions] so that the CF-friendly visualization clients, such as  https://www.giss.nasa.gov/tools/panoply/[Panoply], can visualize these files via Hyrax seamlessly. This feature is realized by setting a BES key _H5.EnableCF_ to _true_.  This is the default BES key setting in the Hyrax distribution.  
 
-== Mappings Between the HDF5 Data Model and DAP2 Data Types
+Four different DAP outputs can be generated via the HDF5 handler. 
 
-The mapping between the HDF5 and HDF-EOS5 data model and DAP2 is
-documented in a NASA Technical Note
-(http://www.esdswg.org/spg/rfc/esds-rfc-017[ESDS-RFC-017]). This note is
-quite detailed; a summary from its appendix is provided below.
+Section <<Highlights>> gives the highlights of these options. The following lists the section that provides detailed information of the four options that generate DAP outputs. 
 
-== Special Characters in HDF Identifiers
+* <<CF Option for DAP4>>
```

Comment:
If by default the `CF Option` is enabled for DAP2/DAP4 (i.e `H5.EnableCF=true`), then I am completely confused by the distinction with `Default Option for DAP4` - are they not the same?  I don't mean to be a pain, but when I read this I find it very confusing, and I think if I am confused our users will be as well. I'm sure there is a difference, but maybe this could be reworded?

After reading the document again I think the issue I am having is that while, by default, the CF option is enabled, in your mind the actual default behavior of the handler is produced when CF is disabled. And that makes the language very tricky. Because the CF option is enabled by default, **_it has become the default behavior_**. And as a result the "_Default Behavior For DAP2|4_" becomes, in effect, the "_Non-CF Compliant Behavior_" or the "_Basic Behavior_".  I think we need to be careful to sort this language out so this is less confusing before we can publish this.

Maybe reworking this opening section so that the Default Behavior is discussed first, and then adding an explanation  that explicitly discusses _that because the CF behavior is so widely used and useful for clients we enable it by default, but it does not actually embody the Default Behavior, but rather an alternate and very useful behavior_ might suffice.

Reply:
I think "discussing the default Behavior" first is a good idea. I will do that and then introduce the CF.  Thanks for the suggestion.

Reply:
Add description how the default and the CF options are from.

Reply:
Your changes look great and totally address my concerns. Thanks for doing that Kent & James!